### PR TITLE
Add filepath.Clean at various path

### PR DESCRIPTION
**Pull request by divyansh42**
Was going through the code and I think it would be better to provide path after passing through `filepath.Clean()`. Therefore, I made the required changes.
Please let me know your thoughts on the same.

### CODE REVIEW

**Comment by rhatdan** [comment]
Thanks @divyansh42 
LGTM
@mtrmac @vrothberg PTAL

**Review by mtrmac — CHANGES_REQUESTED** [review]
Why is this wanted?

`filepath.Clean` changes semantics of the path on Unix, so _by default_ it is undesirable.

File: docker/docker_image_src.go [hunk 1]
```diff
@@ -477,7 +478,7 @@ func (s *dockerImageSource) getOneSignature(ctx context.Context, sigURL *url.URL
 	switch sigURL.Scheme {
 	case "file":
 		logrus.Debugf("Reading %s", sigURL.Path)
-		sigBlob, err := os.ReadFile(sigURL.Path)
+		sigBlob, err := os.ReadFile(filepath.Clean(sigURL.Path))
```
Comment (mtrmac) [inline docker/docker_image_src.go:481]: … and it is actively misleading to log one path and read another.

**Comment by TomSweeneyRedHat** [comment]
I'm hesitant to add this as @mtrmac noted, it does change the path.  That might be good, might not.  However, if we decide to keep it, we should also clean the path in the log too.

**Pull request closed without merging**